src/jit/assertionprop.cpp

```diff
@@ -3188,11 +3210,21 @@ GenTree* Compiler::optAssertionPropGlobal_RelOp(ASSERT_VALARG_TP assertions, Gen
         {
             op1->ChangeOperConst(GT_CNS_INT);
             op1->AsIntCon()->gtIconVal = vnStore->ConstantValue<int>(vnCns);
+
+            if (vnStore->IsVNHandle(vnCns))
```

Comment:
Presumably this case happens only on 32 bit targets?

Reply:
I was also considering not propagating handle constants into trees as they're generally not foldable and end up producing large immediates.